p2p/trust/metric_test.go

```diff
@@ -61,30 +61,32 @@ func TestTrustMetricStopPause(t *testing.T) {
 	tm := NewMetricWithConfig(config)
 
 	// Allow some time intervals to pass and pause
-	time.Sleep(50 * time.Millisecond)
+	tm.NextTimeInterval()
+	tm.NextTimeInterval()
 	tm.Pause()
-	// Give the pause some time to take place
-	time.Sleep(10 * time.Millisecond)
 
 	first := tm.Copy().numIntervals
 	// Allow more time to pass and check the intervals are unchanged
-	time.Sleep(50 * time.Millisecond)
-	assert.Equal(t, first, tm.numIntervals)
+	tm.WaitForTimeIntervalToPass()
+	tm.WaitForTimeIntervalToPass()
+	assert.Equal(t, first, tm.Copy().numIntervals)
 
 	// Get the trust metric activated again
 	tm.GoodEvents(5)
 	// Allow some time intervals to pass and stop
-	time.Sleep(50 * time.Millisecond)
+	tm.NextTimeInterval()
+	tm.NextTimeInterval()
 	tm.Stop()
-	// Give the stop some time to take place
-	time.Sleep(10 * time.Millisecond)
+	// Wait for the stop to take place
+	tm.WaitForStop()
```

Comment:
we can make Stop blocking..

`Wait` is ok too, but it can be simplified imho https://github.com/tendermint/tmlibs/blob/master/common/service.go#L184

btw, is there a specific reason why metric is not a `Service`? i.e. all the things in TM that can be started/stopped follow the same interface and implementation https://github.com/tendermint/tendermint/blob/a2b92c07453f66277f70d0c3979a9dac67bbecea/state/txindex/indexer_service.go#L15